scripts/apps/authoring/authoring/components/CharacterCount.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import classNames from 'classnames';
+import {gettext} from 'core/utils';
+import {cleanHtml} from '../helpers';
+
+interface IProps {
+    limit: number;
+    html: boolean;
+    item: string;
+}
+
+export function getEditorFieldCharactersCount(text: string, html?: boolean) {
+    let input = text || '';
+
+    input = html ? cleanHtml(input) : input;
+    input = input.replace(/\r?\n|\r/g, '');
+
+    return input.length;
+}
+
+export function CharacterCount(props: IProps) {
+    const {limit} = props;
+    const numChars = getEditorFieldCharactersCount(props.item, props.html);
+    const highlightLimit = limit && numChars >= limit;
```

Comment:
Since you are checking if limit is nullish, it should probably be nullable in the interface?